packages/scaffold-config/src/frameworks.ts

```diff
@@ -0,0 +1,579 @@
+
+      module.exports = {
+        component: {
+          devServer,
+        }
+      }`
+      },
+      ts: () => {
+        return dedent`
+      import { defineConfig } from 'cypress'
+      import { devServer } from '@cypress/react/plugins/react-scripts'
+
+      export default defineConfig({
+        component: {
+          devServer,
+        }
+      })`
+      },
+    },
+  },
+  {
+    type: 'crav5',
+    family: 'template',
+    name: 'Create React App (v5)',
+    supportedBundlers: [BUNDLER_WEBPACK_5],
+    packages: [
+      CYPRESS_REACT_LATEST,
+      BUNDLER_WEBPACK_5,
+      CYPRESS_WEBPACK,
+      WEBPACK_DEV_SERVER_4,
+      HTML_WEBPACK_PLUGIN_5,
+    ],
+    defaultPackagePath: '@cypress/react/plugins/react-scripts',
+    glob: '*.{js,jsx,tsx}',
+    category: FRONTEND_FRAMEWORK_CATEGORIES[0],
+    codeGenFramework: CODE_GEN_FRAMEWORKS[0],
+    storybookDep: STORYBOOK_REACT,
+    detectors: [
+      {
+        dependency: 'react-scripts',
+        version: '^5.0.0',
+      },
+    ],
+    config: {
+      js: () => {
+        return dedent`
+      const { devServer } = require('@cypress/react/plugins/react-scripts')
+
+      module.exports = {
+        component: {
+          devServer,
+        }
+      }`
+      },
+      ts: () => {
+        return dedent`
+      import { defineConfig } from 'cypress'
+      import { devServer } from '@cypress/react/plugins/react-scripts'
+
+      export default defineConfig({
+        component: {
+          devServer,
+        }
+      })`
+      },
+    },
+  },
+  {
+    type: 'vueclivue2',
+    name: 'Vue CLI (Vue 2)',
+    family: 'template',
+    supportedBundlers: [BUNDLER_WEBPACK_4],
+    packages: [CYPRESS_VUE_2, CYPRESS_WEBPACK],
+    defaultPackagePath: null,
+    glob: '*.vue',
+    detectors: [
+      {
+        dependency: '@vue/cli-service',
+        version: '^4.0.0',
+      },
+      {
+        dependency: 'vue',
+        version: '^2.0.0',
+      },
+    ],
+    category: FRONTEND_FRAMEWORK_CATEGORIES[1],
+    codeGenFramework: CODE_GEN_FRAMEWORKS[1],
+    storybookDep: STORYBOOK_VUE,
+    config: {
+      js: (bundler: Bundler) => {
+        return dedent`
+      const { devServer } = require('@cypress/webpack-dev-server')
+      const webpackConfig = require('@vue/cli-service/webpack.config')
+
+      module.exports = {
+        component: {
+          devServer,
+          devServerConfig: {
+            webpackConfig
+          }
+        }
+      }
+      `
+      },
+      ts: () => {
+        return dedent`
+      import { defineConfig } from 'cypress'
+      import { devServer } from '@cypress/webpack-dev-server'
+      import webpackConfig from '@vue/cli-service/webpack.config'
+
+      export default defineConfig({
+        component: {
+          devServer,
+          devServerConfig: {
+            webpackConfig
+          }
+        }
+      })`
+      },
+    },
+  },
+  {
+    type: 'vueclivue3',
```

Comment:
Vue-cli v5 is stable which uses webpack 5, fairly new but I'd consider adding a variant or follow up soon after.

Reply:
For sure, maybe you can follow these instructions to add it? https://github.com/cypress-io/cypress/pull/20350/files#diff-8b46b4bb0349c91b80c3f521cb760e26fb444cca40a712b4dd9040384cd6bb0f and update them with anything that's missing so other people can easily add new projects in the future.